### DISCUSSION TITLE
Making my Postal run faster

### DISCUSSION THREAD

**Discussion by navinp** [postalserver/postal#1846] [Help with using Postal]
Hello,

At the moment I'm running my Postal on a 32 GB machine, I would loved it, if there were any tips here, to make my Postal run even faster.

With even faster I mean, faster email delivery. Less than 1 second email delivery. 

Suggestions/tips?

**Comment by willpower232** [comment]
How many workers can you see in `postal status`? That is the main way of increasing throughput.

**Reply by navinp** [reply]
Hello, I have at the moment 5 workers. 

I tried this option last year, but it gave me huge problems.

After increasing the quantity from 5 to 10, I did postal restart, after this, something happened with the memory, it went from normal memory usage to around 20 GB, then to 30 GB, I couldn't use the postal stop feature. 

I shut down the whole machine, restarted, didn't help. Changing the quantity back to 5 didn't help. I had to replace the install with a backup snapshot.

I don't know if this would happen again (or why it happened last year).

But you are saying, this is the only thing I'd need to alter?

**Reply by willpower232** [reply]
the workers are the only thing processing messages backwards and forwards, the queue will hold on to messages and things that need to happen so speed always comes down to how many workers are available to take things out of the queue to do.

You could try with 7 workers to see what happens, if you're hosting the database on the same server as Postal, it is possible that the increased database usage requires even more memory and you might want to look at hosting the databases separately.

**Reply by navinp** [reply]
The database is hosted on the same installation server, will try this on a separate VPS.

**Reply by navinp** [reply]
Hello, I tried changing to 10 workers, but then it says, it failed to have 10 workers and for example only 4 workers are running or 6 workers...

**Reply by navinp** [reply]
In my postal.yml it was like this: 

workers:
  quantity: 5

After browsing here and reading a bit, I changed this now to:

workers:
  quantity: 5
  threads: 5

I tried changing both to 10, but then I get the failed instances (not all 10 workers are running...)

Just tried by changing both to 7, I get this message:

 * worker only has 5 instances (should have 7)

**Reply by navinp** [reply]
At the moment it's taking ~30+ seconds for emails to reach my Gmail inbox... :/

**Reply by willpower232** [reply]
If your server isn't able to run the extra processes perhaps it is running out of resources? You might also consider updating to Postal v2 where the scaling is controlled via docker.
